pkg/operator/encryption/key_controller_encryption_test.go

```diff
@@ -0,0 +1,154 @@
+package encryption
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/diff"
+	"k8s.io/client-go/kubernetes/fake"
+	clientgotesting "k8s.io/client-go/testing"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+const (
+	encryptionSecretKeyDataForTest = "encryption.operator.openshift.io-key"
+)
+
+func TestEncryptionKeyController(t *testing.T) {
+	scenarios := []struct {
+		name                     string
+		initialSecrets           []runtime.Object
+		encryptionSecretSelector metav1.ListOptions
+		targetNamespace          string
+		targetGRs                map[schema.GroupResource]bool
+		validateFunc             func(actions []clientgotesting.Action, targetNamespace string, targetGRs map[schema.GroupResource]bool)
```

Comment:
if you pass `t` as first arg, you can use `t.Fatal` below.

Reply:
Oh, you even have to. The `t` bound in these scenario funcs is not the subtest one now.

Reply:
good catch, thx!